### DISCUSSION TITLE
[Config Support]: Reccording triggered  by external trigger.

### DISCUSSION THREAD

**Discussion by grostim** [blakeblackshear/frigate#15340] [Config Support]
### Describe the problem you are having

my door camera is triggered by any pedestrian walking down the street. I don't want that, thus i have deactivated the automatic record of "person".
However, i would like to trigger a record each time someone is pressing the doorbell.
I can get the doorbell event through mqtt.
How could i trigger the recording based on this event ?

### Version

0.14

### Frigate config file

```yaml
N/A
```

### Relevant Frigate log output

```shell
N/A
```

### Relevant go2rtc log output

```shell
N/A
```

### Frigate stats

```json
N/A
```

### Operating system

Debian

### Install method

Docker Compose

### docker-compose file or Docker CLI command

```yaml
N/A
```

### Object Detector

Coral

### Screenshots of the Frigate UI's System metrics pages

N/A

### Any other information that may be helpful

N/A

**Comment by hawkeye217** [comment]
Rather than disabling recording for `person`, you could set a `min_area` [object filter](https://docs.frigate.video/configuration/object_filters) to prevent smaller persons from triggering your recording, eg:

```
objects:
  filters:
    person:
      min_area: 10000
```

If you do want to trigger on a doorbell press, you can create a manual event through the HTTP API. See [the docs](https://docs.frigate.video/integrations/api#post-apieventscamera_namelabelcreate).

**Comment by AbDhops** [comment]
This is exactly what I am trying to do (as written by OP). I want to store recording, ONLY when door bell is pressed. So based on doorbell "event", I am starting the recording using API. And based on certian conditions I am stopping the recording again using the API (all in Node Red)

My start API looks like following (it does not include `duration`)
```
"sub_label": "Door Bell Pressed",
"include_recording": true
```

I can see in log, that event is started. 
```
frigate  | 2024-12-18 11:15:11.225258099  <MY_NODE_RED_IP> - - [18/Dec/2024:11:15:11 +0100] "POST /api/events/Doorbell/doorbell/create HTTP/1.1" 200 95 "-" "got (https://github.com/sindresorhus/got)" "-"
```

After some time, I do send stop command using http PUT request

```
http://192.168.100.32:5000/api/events/1734516809.280526-dz1fnv/end
```

I can see below in logs. 

```
frigate  | 2024-12-18 11:15:51.321285834  [2024-12-18 11:15:51] frigate.review.maintainer      ERROR   : Event with ID 1734516911.070126-jk5brt has a set duration and can not be ended manually.
frigate  | 2024-12-18 11:15:51.326202666  <MY_NODE_RED_IP> - - [18/Dec/2024:11:15:51 +0100] "PUT /api/events/1734516911.070126-jk5brt/end HTTP/1.1" 200 55 "-" "got (https://github.com/sindresorhus/got)" "-"

```

I tested this with multiple time duration, and could see the recordings of correct legnth. 
What I wonder is,
1. The error in the log after sending the stop command
```
frigate  | 2024-12-18 11:15:51.321285834  [2024-12-18 11:15:51] frigate.review.maintainer      ERROR   : Event with ID 1734516911.070126-jk5brt has a set duration and can not be ended manually.
```

2. The link to _got repo_ in the logs whenever I do POST / PUT request (`got (https://github.com/sindresorhus/got)`). Is this correct ?
`

**Reply by NickM-27** [reply]
Please read the docs, if you want the event to be an indefinite duration you need to send null as the duration
